pkg/drift_client/drift_client.py

```diff
@@ -192,52 +213,60 @@ def package_handler(message):
+        Args:
+            topic: MQTT topic
+            payload: Stringified data, defaults to None
 
         # Examples
             >>> client = DriftClient("127.0.0.1", "PASSWORD")
-            >>> client.publish_data("topic-2", "hello")
+            >>> client.publish_data("topic-2", b"hello")
         """
         if not self.__mqtt_client.is_connected():
             self.__mqtt_client.connect()
             self.__mqtt_client.loop_start()
 
         self.__mqtt_client.publish(topic, payload)
 
-    def _convert_type(self, timestamp: Union[float, datetime, str]) -> str:
-        if isinstance(timestamp, str):
-            return datetime.fromisoformat(timestamp).strftime("%Y-%m-%d %H:%M:%S")
-        if isinstance(timestamp, float):
-            return datetime.fromtimestamp(timestamp).strftime("%Y-%m-%d %H:%M:%S")
-        if isinstance(timestamp, datetime):
-            return timestamp.strftime("%Y-%m-%d %H:%M:%S")
-        raise TypeError("Timestamp must be str, float or datetime")
-
-    def _get_topic_data(self, topic: str, start: str, stop: str) -> List[str]:
-        """Returns list of history data from initialised Device
-
+    def get_metrics(
+        self,
+        topic: str,
+        start: Union[float, datetime, str],
+        stop: Union[float, datetime, str],
+        names: Optional[List[str]] = None,
+    ) -> List[Dict[str, Any]]:
+        """Reads history metrics from timeseries database
+            >>> client = DriftClient("127.0.0.1", "PASSWORD")
+            >>> client.get_metrics("topic", "2022-02-03 10:00:00",
+            >>>    "2022-02-03 10:00:10", names=["status", "field"])
+            >>> #=> [{"status": 0, "field": 0.1231}, ....]
         Args:
-            topics: Topic name, e.g. `"sensor-1"`
+            topic: MQTT topic
             start: Begin of request timeframe,
-                Format: `2022-02-07 10:00:00`
+                Format: ISO string, datetime or float timestamp
             stop: End of request timeframe,
-                Format: `2022-02-07 10:00:00`
-
-        Returns:
-            List with item names available
-        :rtype: List[str]
+                Format: ISO string, datetime or float timestamp
         """
-        data = []
+
+        start = _convert_type(start)
+        stop = _convert_type(stop)
+
+        aligned_data = {}
         influxdb_values = self.__influx_client.query_data(
-            topic, start, stop, field="status"
+            topic, start, stop, fields=names
         )
 
-        if influxdb_values:
-            for timestamp, _ in influxdb_values:
-                data.append(f"{topic}/{int(timestamp * 1000)}.dp")
+        for field, values in influxdb_values.items():
+            for dt, value in values:
+                if dt not in aligned_data:
+                    aligned_data[dt] = {}
+
+                aligned_data[dt][field] = value
+
+        data = []
+        for dt, fields in aligned_data.items():
```

Comment:
dt name is confusing. what does it mean, data time? what about timestamp?

Reply:
ah, datetime. makes sense

Reply:
`timestamp` is a  name of a module in Python like `time` and `datetime`. I could use _time I think, but it is also tricky  because people use is to mark variables that maybe unused.... .